identity-iota/src/chain/diff_chain.rs

```diff
@@ -316,4 +326,45 @@ mod test {
       assert!(chain.try_push_diff(diff).is_ok());
     }
   }
+
+  #[test]
+  fn test_check_valid_addition_updating_capability_invocation() {
```

Comment:
I think we need to test a few more cases for completeness since this only checks removing a signing method. Most important is the case where a `DiffMessage` attempts to update a signing method that is a verification method with a referenced capability invocation relationship, rather than being embedded as one.

E.g.
```rust
  #[test]
  fn test_check_valid_addition_rejects_altering_referenced_signing_method() {
    // Create initial document.
    let keypair: KeyPair = KeyPair::new_ed25519().unwrap();
    let mut document: IotaDocument = IotaDocument::new(&keypair).unwrap();

    // Replace the embedded capability invocation signing method with a reference to a
    // verification method entry.
    let signing_method: IotaVerificationMethod = document.default_signing_method().unwrap().clone();
    let signing_method_id: IotaDIDUrl = signing_method.id();
    document.remove_method(signing_method.id()).unwrap();
    document
      .insert_method(signing_method, MethodScope::VerificationMethod)
      .unwrap();
    assert!(document
      .attach_method_relationship(signing_method_id, MethodRelationship::CapabilityInvocation)
      .unwrap());

    document
      .sign_self(keypair.private(), &document.default_signing_method().unwrap().id())
      .unwrap();
    document.set_message_id(MessageId::new([8; 32]));
    let chain: DocumentChain = DocumentChain::new(IntegrationChain::new(document.clone()).unwrap());

    // INVALID: alter a verification method that has an attached capability invocation relationship
    //          in a diff update.
    let mut new_document: IotaDocument = document.clone();
    // Replace the public key data.
    unsafe {
      let updated_method: &mut VerificationMethod = new_document
        .as_document_mut()
        .verification_method_mut()
        .head_mut()
        .unwrap();
      *updated_method.key_data_mut() = MethodData::new_multibase([3u8; 32]);
    }
    new_document.set_previous_message_id(*chain.integration_message_id());
    new_document.set_updated(Timestamp::now_utc());

    let mut diff_msg: DiffMessage =
      DiffMessage::new(&document, &new_document, *chain.integration_message_id()).unwrap();
    document
      .sign_data(
        &mut diff_msg,
        keypair.private(),
        document.default_signing_method().unwrap().id(),
      )
      .unwrap();
    diff_msg.set_message_id(*chain.diff_message_id());

    let valid_addition_error: Error =
      DiffChain::check_valid_addition(&diff_msg, &document, chain.integration_message_id()).unwrap_err();
    assert!(matches!(
      valid_addition_error,
      Error::ChainError {
        error: "diff cannot alter update signing methods"
      }
    ));
  }
```

And here's two more cases covering adding and altering the signing method (could be combined or use helper functions to prevent duplicate test code everywhere):

```rust
  #[test]
  fn test_check_valid_addition_rejects_adding_signing_method() {
    // Create initial document.
    let keypair: KeyPair = KeyPair::new_ed25519().unwrap();
    let mut document: IotaDocument = IotaDocument::new(&keypair).unwrap();
    document
      .sign_self(keypair.private(), &document.default_signing_method().unwrap().id())
      .unwrap();
    document.set_message_id(MessageId::new([8; 32]));
    let chain: DocumentChain = DocumentChain::new(IntegrationChain::new(document.clone()).unwrap());

    // INVALID: add a capability invocation method in a diff update.
    let mut new_document: IotaDocument = document.clone();
    let new_signing_method: IotaVerificationMethod = IotaVerificationMethod::from_did(
      new_document.id().clone(),
      keypair.type_(),
      keypair.public(),
      "new-signing-key",
    )
    .unwrap();
    new_document
      .insert_method(new_signing_method, MethodScope::capability_invocation())
      .unwrap();
    new_document.set_previous_message_id(*chain.integration_message_id());
    new_document.set_updated(Timestamp::now_utc());

    let mut diff_msg: DiffMessage =
      DiffMessage::new(&document, &new_document, *chain.integration_message_id()).unwrap();
    document
      .sign_data(
        &mut diff_msg,
        keypair.private(),
        document.default_signing_method().unwrap().id(),
      )
      .unwrap();
    diff_msg.set_message_id(*chain.diff_message_id());

    let valid_addition_error: Error =
      DiffChain::check_valid_addition(&diff_msg, &document, chain.integration_message_id()).unwrap_err();
    assert!(matches!(
      valid_addition_error,
      Error::ChainError {
        error: "diff cannot alter update signing methods"
      }
    ));
  }

  #[test]
  fn test_check_valid_addition_rejects_altering_signing_method() {
    // Create initial document.
    let keypair: KeyPair = KeyPair::new_ed25519().unwrap();
    let mut document: IotaDocument = IotaDocument::new(&keypair).unwrap();
    document
      .sign_self(keypair.private(), &document.default_signing_method().unwrap().id())
      .unwrap();
    document.set_message_id(MessageId::new([8; 32]));
    let chain: DocumentChain = DocumentChain::new(IntegrationChain::new(document.clone()).unwrap());

    // INVALID: alter a capability invocation method in a diff update.
    let mut new_document: IotaDocument = document.clone();
    // Replace the public key data.
    unsafe {
      match new_document
        .as_document_mut()
        .capability_invocation_mut()
        .head_mut()
        .unwrap()
      {
        MethodRef::Embed(method) => {
          *method.key_data_mut() = MethodData::new_multibase([3u8; 32]);
        }
        MethodRef::Refer(_) => unreachable!(),
      };
    }
    new_document.set_previous_message_id(*chain.integration_message_id());
    new_document.set_updated(Timestamp::now_utc());

    let mut diff_msg: DiffMessage =
      DiffMessage::new(&document, &new_document, *chain.integration_message_id()).unwrap();
    document
      .sign_data(
        &mut diff_msg,
        keypair.private(),
        document.default_signing_method().unwrap().id(),
      )
      .unwrap();
    diff_msg.set_message_id(*chain.diff_message_id());

    let valid_addition_error: Error =
      DiffChain::check_valid_addition(&diff_msg, &document, chain.integration_message_id()).unwrap_err();
    assert!(matches!(
      valid_addition_error,
      Error::ChainError {
        error: "diff cannot alter update signing methods"
      }
    ));
  }
``